### PR TITLE
[MM-57720] Convert save_button.tsx to function component

### PR DESCRIPTION
**Pull request by tasnim0tantawi**
#### Summary
Convert ./component/save_button.tsx from class to function component. 
Addressed issue https://github.com/mattermost/mattermost/issues/26688

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/save_button.tsx [hunk 1]
```diff
@@ -1,10 +1,10 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import React from 'react';
-import {FormattedMessage} from 'react-intl';
+import React from "react";
```
Comment (larkox) [inline webapp/channels/src/components/save_button.tsx:4]: Seems like your IDE is automatically correcting some things and conflicting with our intl rules. In particular, single quotes vs double quotes.

You can fix it really simple if you use `npm run fix`, but your IDE may still be updating this for you, making future development annoying :sweat_smile: Take a look at that.

File: webapp/channels/src/components/save_button.tsx [hunk 2]
```diff
@@ -15,67 +15,48 @@ type Props = {
...
+        <button
+            type="submit"
+            data-testid="saveSetting"
+            id="saveSetting"
+            className={className}
+            disabled={disabled}
+            {...props}
+        >
+            <LoadingWrapper loading={saving} text={savingMessage}>
+                {defaultMessage}
```
Comment (larkox) [inline webapp/channels/src/components/save_button.tsx:56]: Why removing the `span`?

**Review by larkox — CHANGES_REQUESTED** [review]
Great work. Just one comment, and also check out the linting.

File: webapp/channels/src/components/save_button.tsx (hunk 1 above)
Comment (tasnim0tantawi) [reply webapp/channels/src/components/save_button.tsx]: Fixed the single quote and linting

**Review by larkox — APPROVED** [review]
Looks good to me. Thanks!

**Comment by larkox** [comment]
@tasnim0tantawi There seem to be some issue with the types. I don't understand why, `saving` was understood before as an optional value (even though the type definition doesn't explicitly declare it as optional). Just setting saving to be optional should be enough, I think. Can you check it?

**Comment by tasnim0tantawi** [comment]
> @tasnim0tantawi There seem to be some issue with the types. I don't understand why, `saving` was understood before as an optional value (even though the type definition doesn't explicitly declare it as optional). Just setting saving to be optional should be enough, I think. Can you check it?

Done, set the default value for `saving` to `false`

File: webapp/channels/src/components/save_button.tsx [hunk 3]
```diff
@@ -1,81 +1,68 @@
-// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
-// See LICENSE.txt for license information.
-
```
Comment (larkox) [inline webapp/channels/src/components/save_button.tsx:3]: Removing this is going to create a lint issue :sweat_smile:

**Review by larkox — CHANGES_REQUESTED** [review]
Just one minor unintended change, I think

File: webapp/channels/src/components/save_button.tsx (hunk 3 above)
Comment (tasnim0tantawi) [reply webapp/channels/src/components/save_button.tsx]: Sorryy by mistake my IDE is being quite wierd. Added it back

**Review by larkox — APPROVED** [review]

**Comment by larkox** [comment]
@tasnim0tantawi There seems to be some snapshots that need to be updated. Just running `npm run test -- -u` should do the trick.

File: webapp/channels/src/components/save_button.tsx [hunk 4]
```diff
@@ -7,75 +7,65 @@ import {FormattedMessage} from 'react-intl';
...
-                    loading={saving}
-                    text={savingMessage}
-                >
-                    <span>{defaultMessage}</span>
-                </LoadingWrapper>
-            </button>
-        );
-    }
-}
+                <span> {defaultMessage} </span>
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/save_button.tsx:65]:
```suggestion
                <span>{defaultMessage}</span>
```

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]
Just a small nit

File: webapp/channels/src/components/save_button.tsx (hunk 4 above)
Comment (tasnim0tantawi) [reply webapp/channels/src/components/save_button.tsx]: Done

**Review by M-ZubairAhmed — APPROVED** [review]
This looks good :shipit: Thank you

**Comment by tasnim0tantawi** [comment]
> @tasnim0tantawi There seems to be some snapshots that need to be updated. Just running `npm run test -- -u` should do the trick.

I have tried to run this command, and `npm run test-ci -- -u`  as well. However, snapshots are not being updated. But, the save_button.tsx snapshots are passing fine. Need your assistance.

**Comment by larkox** [comment]
@tasnim0tantawi The idea was to run the `test` command, not the `test-ci` one. I just ran it for you and updated the snapshots :smile:

**Comment by tasnim0tantawi** [comment]
> @tasnim0tantawi The idea was to run the `test` command, not the `test-ci` one. I just ran it for you and updated the snapshots 😄

I tried both, didnt work sadly. Thanks so much !

**Comment by larkox** [comment]
@tasnim0tantawi Thank you very much for the contribution!

**Pull request merged**